app/version.go

```diff
@@ -0,0 +1,113 @@
+package app
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/TIBCOSoftware/flogo-cli/cli"
+	toml "github.com/pelletier/go-toml"
+)
+
+var optVersion = &cli.OptionInfo{
+	Name:      "version",
+	UsageLine: "version",
+	Short:     "displays the version of flogo cli",
+	Long: `Get the current version number of the cli.
+
+`,
+}
+
+type rawLock struct {
+	SolveMeta solveMeta          `toml:"solve-meta"`
+	Projects  []rawLockedProject `toml:"projects"`
+}
+
+type solveMeta struct {
+	InputsDigest    string `toml:"inputs-digest"`
+	AnalyzerName    string `toml:"analyzer-name"`
+	AnalyzerVersion int    `toml:"analyzer-version"`
+	SolverName      string `toml:"solver-name"`
+	SolverVersion   int    `toml:"solver-version"`
+}
+
+type rawLockedProject struct {
+	Name     string   `toml:"name"`
+	Branch   string   `toml:"branch,omitempty"`
+	Revision string   `toml:"revision"`
+	Version  string   `toml:"version,omitempty"`
+	Source   string   `toml:"source,omitempty"`
+	Packages []string `toml:"packages"`
+}
+
+const lockName = "Gopkg.lock"
+
+func init() {
+	CommandRegistry.RegisterCommand(&cmdVersion{option: optVersion})
+}
+
+type cmdVersion struct {
+	option *cli.OptionInfo
+}
+
+// HasOptionInfo implementation of cli.HasOptionInfo.OptionInfo
+func (c *cmdVersion) OptionInfo() *cli.OptionInfo {
+	return c.option
+}
+
+// Exec implementation of cli.Command.Exec
+func (c *cmdVersion) AddFlags(fs *flag.FlagSet) {
+	//op op
+}
+
+// Exec implementation of cli.Command.Exec
+func (c *cmdVersion) Exec(args []string) error {
+
+	cmd := exec.Command("git", "describe", "--tags")
+	cmd.Dir = filepath.Join(os.Getenv("GOPATH"), "src", "github.com", "TIBCOSoftware", "flogo-cli")
+	cmd.Env = append(os.Environ())
+
+	out, err := cmd.Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+	re := regexp.MustCompile("\\n")
+	fc := re.ReplaceAllString(string(out), "")
+
+	line := fmt.Sprintf("flogo cli version [%s]\n", fc)
+	fmt.Fprint(os.Stdout, line)
+
+	appDir, err := os.Getwd()
+
+	if err != nil {
+		fmt.Fprint(os.Stderr, "Error: Unable to determine working directory\n\n")
+		os.Exit(2)
+	}
+
+	project := SetupExistingProjectEnv(appDir)
+
+	config, err := toml.LoadFile(filepath.Join(project.GetAppDir(), lockName))
+
+	if err != nil {
+		fmt.Println("Error ", err.Error())
+	} else {
+		raw := rawLock{}
+		err := config.Unmarshal(&raw)
+		if err != nil {
+			fmt.Printf("Unable to parse the lock as TOML")
+		}
+
+		for _, v := range raw.Projects {
+			if strings.Contains(v.Name, "flogo") {
```

Comment:
i think we may want to specifically check for `github.com/TIBCOSoftware/flogo-contrib` and `github.com/TIBCOSoftware/flogo-lib`. Many people fork those repos and may have their own activities/triggers, etc and the cli would display the repo twice with different versions and that may be confusing... for example:

Your project uses flogo-contrib version 0.5.3
Your project uses flogo-contrib version 0.5.2

Reply:
or optionally, don't split the string below :)

Reply:
also perhaps use EqualFold just in the event the case doesn't match?

Reply:
Hmm... I see your point. So I've updated the code to show the rest of the import name as well (because technically the package could come from a non-github source as well) so the output is now
```
λ flogo version
flogo cli version [v0.5.3-9-g0f0c548]
Your project uses github.com/TIBCOSoftware/flogo-contrib version v0.5.3
Your project uses github.com/TIBCOSoftware/flogo-lib version v0.5.3
Your project uses github.com/retgits/flogo-components branch master and revision e55b8f033bd56e28c51d24de735ff522ae8f733d
```
So it shows the full path of the import (which makes sense) and it shows either version or branch/revision

Using EqualFold would only check for an exact string and not a substring, so the function `caseInsensitiveContains` takes care of that